src/rules/eva-voice-biometric.js

```diff
@@ -0,0 +1,165 @@
+/**
+ *	@title EVA Voice Biometric connector
+ *	@overview EVA Voice Biometric connector rule for Auth0 enables voice enrolment and verification as a second factor
+ *	@gallery true
+ *	@category marketplace
+ *
+ *	EVA WEB is a second-factor voice biometric solution designed for use with a range of web browsers that support audio capture.
+ *	The biometric verification process is intended to be part of an authentication flow to enhance security, privacy and ease-of-use.
+ *	EVA WEB is an offering from Auraya Systems.
+ *
+ *	The voice tokens used with EVA WEB are predominantly digit strings, although phrases are also supported.
+ *	The usual verification process will usually prompt the user to read out a sequence of random digits,
+ *	a sequence of consecutive digits, and optionally their phone number.
+ *
+ *	The first time a user is directed to EVA WEB where they will be invited to enrol their voice.
+ *	This process captures between 3 and 8 utterances depending on the required voice tokens, and includes a biometric “Opt-In” consent step.
+ *	At the conclusion of enrolment they will be redirected back to the authentication flow.
+ *
+ *	Configuration Items:
+ *	AURAYA_URL = EVA endpoint, typically: https://eva-web.mydomain.com/server/oauth
+ *	AURAYA_CLIENT_ID = JWT client id on the EVA server (and this server)
+ *	AURAYA_CLIENT_SECRET = JWT client secret on the EVA server (and this server)
+ *	AURAYA_ISSUER = this app (or "issuer")
+ *
+ *	AURAYA_RANDOM_DIGITS = true|false whether to prompt for random digits
+ *	AURAYA_COMMON_DIGITS = true|false whether to prompt for common digits
+ *	AURAYA_PERSONAL_DIGITS = a user.user_metadata field that contains digits such as phone_number, or blank
+ *
+ *	AURAYA_DEBUG = true|false (or blank) controls debug output
+ */
+function evaVoiceBiometric(user, context, callback) {
+  if (
+    typeof configuration.AURAYA_DEBUG !== 'undefined' &&
+    configuration.AURAYA_DEBUG === 'true'
+  ) {
+    console.log(user);
+    console.log(context);
+    console.log(configuration);
+  }
+
+  if (
+    !configuration.AURAYA_COMMON_DIGITS ||
+    !configuration.AURAYA_RANDOM_DIGITS ||
+    !configuration.AURAYA_PERSONAL_DIGITS ||
```

Comment:
Are these actually required, as in the Rule will not run properly without them, or will EVA provide a default?

Reply:
I'll make them optional with sensible defaults.